src/httpify/client.go

```diff
@@ -110,14 +102,38 @@ func (client *HTTPClient) do(method, path string, body interface{}, headers map[
 		req.Header.Add(label, value)
 	}
 
-	client.limit.Wait()
-	resp, err := client.client.Do(req)
-	if err, ok := err.(net.Error); ok && err.Timeout() {
-		return nil, errClientTimeout
-	} else if err != nil && strings.Contains(err.Error(), "no such host") {
-		return nil, errConnectionIssue
+	return client.doWithRetry(req, body)
+}
+
+func (client *HTTPClient) doWithRetry(req *http.Request, body interface{}) (*http.Response, error) {
+	attempt := 0
+	for {
+		// reset the body when non-nil for every request (rewind)
+		if body != nil {
+			data, err := json.Marshal(body)
+			if err != nil {
+				return nil, err
+			}
+			req.Body = ioutil.NopCloser(bytes.NewBuffer(data))
+		}
+
+		client.limit.Wait()
+		resp, err := client.client.Do(req)
+		if err == nil && resp.StatusCode >= 100 && resp.StatusCode <= 428 {
```

Comment:
Why do we choose 428 as the cutoff here?

Reply:
Because 429 is too many requests and I need that response for the `retry-after` header.

Reply:
For clarity, for perhaps we can define a constant such as `HTTPTooManyRequests` and write this as `resp.StatusCode <  HTTPTooManyRequests`